hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -137,11 +136,14 @@ class MergeOnReadSnapshotRelation(val sqlContext: SQLContext,
   }
 
   def buildFileIndex(filters: Array[Filter]): List[HoodieMergeOnReadFileSplit] = {
-
-    val fileStatuses = if (globPaths.isDefined) {
+    // Get all partition paths
+    val partitionPaths = if (globPaths.isDefined) {
       // Load files from the global paths if it has defined to be compatible with the original mode
       val inMemoryFileIndex = HoodieSparkUtils.createInMemoryFileIndex(sqlContext.sparkSession, globPaths.get)
-      inMemoryFileIndex.allFiles()
+      val fsView = new HoodieTableFileSystemView(metaClient,
+        metaClient.getActiveTimeline.getCommitsTimeline
+          .filterCompletedInstants, inMemoryFileIndex.allFiles().toArray)
+      fsView.getLatestBaseFiles.iterator().asScala.toList.map(_.getFileStatus.getPath.getParent)
```

Comment:
if we only have log files.  inMemoryFileIndex.allfiles will be empty, since spark will filter .log
fsView.getLatestBaseFiles.iterator().asScala.toList.map(_.getFileStatus.getPath.getParent)  will return a empty partitionPaths.  then buildFileIndex will return HoodieMergeOnReadFileSplit, nothings will be read

Reply:
Yes, currently I have not support reading-log only table  for the `globPaths` query which users must specify some `*` in the query path and the partition prune is also not work.
 In the long way,  we should recommend the non-globalPath query which use the `HoodieFileIndex` to list files and partitions. User do not need to specify the `*`  in the query path and can also have better performance from the partition prune and meta table list.

Reply:
Thank you for your explanation， LGTM

Reply:
Is there a Jira issue about the plan?

Reply:
Currently we have already support the `non-globalPaths ` query way by https://github.com/apache/hudi/pull/2651, I will add some doc to introduce it in 0.9 release.

Reply:
Oh nice. Didn't follow Spark related changes recently. Seems like it has a lot of new features landed.